### PR TITLE
feat(loki/src/k8s): not restart tailers in `loki.source.kubernetes` component by above-average time deltas if K8s version is > 1.29.0

### LINKED ISSUE #6250: [loki.source.kubernetes] decrease logLevel of log msg
### What's wrong?

I am receiving the follow log message multiple times a second.  This seems to be expected behavior from this component, however it's also expected behavior from my (dozens of ceph osd) pod.  I feel the appropriate solution here would be to decrease this log to `level=debug`, or alternatively to allow configuration of it somehow.

```
ts=2024-01-17T01:28:07.837938857Z level=info msg="have not seen a log line in 3x average time between lines, closing and re-opening tailer" target=ceph/rook-ceph-osd-16-6d87ddf895-kghz8:osd component=loki.source.kubernetes.allPods rolling_average=2s time_since_last=6.739761385s
```

### Steps to reproduce

NA

### System information

_No response_

### Software version

docker.io/grafana/agent:v0.39.0

### Configuration

```text
Running agent using the helm chart in Daemonset Flow mode
```

### Logs

```text
NA
```

### PR DESCRIPTION
**Pull request by hainenber**
#### PR Description
Detect K8s version to decide whether to kick off goroutine to restart tailers by above-average deltas. Helps decreasing extraneous logs and might be a tiny bit of performance.

#### Which issue(s) this PR fixes

Fixes grafana/alloy#264 

#### PR Checklist

- [x] CHANGELOG.md updated

### CODE REVIEW

**Review by rfratto — APPROVED** [review]
Sorry this one got lost! LGTM, if you have time to fix the merge conflicts let me know, otherwise I'll push a commit to fix them once I get a chance.

**Comment by hainenber** [comment]
I've resolved the merge conflict :D

**Review by tpaschalis — APPROVED** [review]
Sorry that this one slipped through the cracks @hainenber! I've taken the liberty to fix merge conflicts so we can merge and cherry-pick over to the Alloy repo.

Again, thank you!

**Pull request merged**
